src/pages/pupils/programmes/[programmeSlug]/units.tsx

```diff
@@ -182,10 +183,12 @@ export const getStaticProps: GetStaticProps<
 
       const secondSectionLength = secondUnitSection.units.length;
 
+      const iconSlug = `subject-${subjectSlug}`;
+
       const firstUnitSection: UnitsSectionData = {
         units: optionalityUnits,
         phase,
-        icon: `subject-${subjectSlug}`,
+        icon: isValidIconName(iconSlug) ? iconSlug : undefined,
```

Comment:
we currently don't have all the subject icons for cycle 2 so I've added this handler.